### PR TITLE
chore(dbt-fal): Update for 1.4 dbt-core support 

### PR DESCRIPTION
**Pull request by chamini2**
### Integration tests

Adapter to test:

- postgres
- bigquery
- duckdb
- snowflake
- redshift

Python version to test:

- 3.8

- 3.10

### CODE REVIEW

**Comment by mederka** [comment]
This is so strange:
```
[df64af] [builder]  [debug]   Installing requirements: dbt-core==1.4.1, dbt-redshift==1.4.0, dbt-fal[redshift]==1.3.13a0, isolate==0.8.0
[df64af] [builder]  [debug]   Collecting dbt-core==1.4.1
[df64af] [builder]  [debug]   ERROR: Could not find a version that satisfies the requirement dbt-fal[redshift]==1.3.13a0 (from versions: 1.3.0rc1, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.8, 1.3.9, 1.3.10, 1.3.11, 1.3.12)
```
why is it trying to install two different versions of dbt-redshift?

**Comment by chamini2** [comment]
> This is so strange:
> 
> ```
> [df64af] [builder]  [debug]   Installing requirements: dbt-core==1.4.1, dbt-redshift==1.4.0, dbt-fal[redshift]==1.3.13a0, isolate==0.8.0
> [df64af] [builder]  [debug]   Collecting dbt-core==1.4.1
> [df64af] [builder]  [debug]   ERROR: Could not find a version that satisfies the requirement dbt-fal[redshift]==1.3.13a0 (from versions: 1.3.0rc1, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.8, 1.3.9, 1.3.10, 1.3.11, 1.3.12)
> ```
> 
> why is it trying to install two different versions of dbt-redshift?

Those are all different packages. But you did point to the line where we have a problem. We are not using the `pip install '/Absolute/path/to/adapter[redshift]'` format, and we should

**Comment by mederka** [comment]
You're right, @chamini2, it's probably my sleeping time.

**Comment by mederka** [comment]
```
dbt-redshift 1.3.0 depends on dbt-core~=1.3.0
dbt-fal[cloud,redshift] 1.3.13a0 depends on dbt-core<1.5 and >=1.4.0
```
I guess it should be trying to install dbt-redshift 1.4, right?

File: .github/workflows/test_integration_adapter.yml [hunk 1]
```diff
@@ -33,13 +33,15 @@ jobs:
           - redshift
           # enable with next release of dbt-trino
           # - trino
-        python: ["3.7", "3.8", "3.11"]
         dbt_version:
-          # TODO: add 1.4 when it's supported
-          - "1.3.*"
+          - "1.4.*"
+        python:
```
[the remark this answers is not in the document]
Comment (chamini2) [reply .github/workflows/test_integration_adapter.yml:38]: Just to avoid running so many. We can, though.

**Review by isidentical — APPROVED** [review]
Looks great!

**Pull request merged**